mathics_django/web/media/js/mathics.js

```diff
@@ -1,829 +1,881 @@
+		    const foreignObject = math.parentNode.parentNode.parentNode,
+			  dimensions = math.getDimensions();
 
-					const ox = parseFloat(foreignObject.getAttribute('ox')),
-						oy = parseFloat(foreignObject.getAttribute('oy')),
-						width = dimensions.width + 4,
-						height = dimensions.height + 4;
+		    const ox = parseFloat(foreignObject.getAttribute('ox')),
+			  oy = parseFloat(foreignObject.getAttribute('oy')),
+			  width = dimensions.width + 4,
+			  height = dimensions.height + 4;
 
-					let x = parseFloat(foreignObject.getAttribute('x').substr()),
-						y = parseFloat(foreignObject.getAttribute('y'));
+		    let x = parseFloat(foreignObject.getAttribute('x').substr()),
+			y = parseFloat(foreignObject.getAttribute('y'));
 
-					x = x - width / 2.0 - ox * width / 2.0;
-					y = y - height / 2.0 + oy * height / 2.0;
+		    x = x - width / 2.0 - ox * width / 2.0;
+		    y = y - height / 2.0 + oy * height / 2.0;
 
-					foreignObject.setAttribute('x', x + 'px');
-					foreignObject.setAttribute('y', y + 'px');
-				}
-			});
-	});
+		    foreignObject.setAttribute('x', x + 'px');
+		    foreignObject.setAttribute('y', y + 'px');
+		}
+	    });
+    });
 }
 
 function setResult(list, results) {
-	const resultList = document.createElement('ul');
-	resultList.className = 'out';
-	// we'll just show if it have children
-	resultList.style.display = 'none';
+    // Formats DJango cell "results" and add that onto "list".
 
-	results.forEach((result) => {
-		result.out.forEach((out) => {
-			const li = document.createElement('li');
-			li.className = out.message ? 'message' : 'print';
+    const resultList = document.createElement('ul');
+    resultList.className = 'out';
 
-			if (out.message) {
-				li.innerHTML += out.prefix + ': ';
-			}
+    resultList.style.display = 'none';
 
-			li.appendChild(createLine(out.text));
+    const li = document.createElement('li');
+    li.className = 'out';
 
-			resultList.appendChild(li);
-		});
+    if (results.length == 1 && results[0].form == "Syntax Error") {
+	const result = results[0];
+	const out = result.out[0];
 
-		if (result.result) {
-			const li = document.createElement('li');
-			li.className = 'result';
-			li.appendChild(createLine(result.result));
+	// Syntax Error prefix or code
+	const bold = document.createElement('b');
+	bold.className = "message"
+	bold.innerHTML = out.prefix + ":"
+	resultList.appendChild(bold);
 
-			resultList.appendChild(li);
-			resultList.style.display = 'block';
-		}
 
-		if (result.out.length) {
-			resultList.style.display = 'block';
+	// Syntax Error message
+	const pre = document.createElement('pre');
+	pre.innerHTML = out.text;
+
+	// Add this to a list element in a block (which is CSS styled with a nice
+	// frame around it.
+	resultList.appendChild(pre);
+	resultList.style.display = 'block';
+	li.appendChild(resultList);
+	list.appendChild(li);
+    } else if (results.length == 1 && results[0].form == "Python Exception") {
```

Comment:
This is the other major tag detection change here where we do a rudimentary format of a traceback.